main.py

```diff
@@ -0,0 +1,102 @@
+            for proxy in req.split("\n"):
+                proxy = proxy.strip()
+                proxy = f"https://{proxy}"
+                open("./data/proxies.txt", "a").write(f"{proxy}\n")
+
+    def generate(this, scrape=None):
+        if scrape == True:
+            this.__proxies__()
+        else:
+            pass
+
+        os.system("clear")
+        for _ in range(int(this.codes)):
+            try:
+                if this.proxies == True:
+                    prox = {
+                        "https": random.choice(
+                            open("./data/proxies.txt", "r").read().splitlines()
+                        )
+                    }
+                else:
+                    prox = None
+
+                if this.type == "boost":
+                    code = "".join(
+                        [
+                            random.choice(string.ascii_letters + string.digits)
+                            for i in range(24)
+                        ]
+                    )
+                else:
+                    code = "".join(
+                        [
+                            random.choice(string.ascii_letters + string.digits)
+                            for i in range(16)
+                        ]
+                    )
+                req = this.session.get(
+                    f"https://discordapp.com/api/v6/entitlements/gift-codes/{code}",
+                    proxies=prox,
+                    timeout=10,
+                ).status_code
+                if req == 200:
+                    print(
+                        f"{Fore.GREEN}[{strftime('%H:%M', localtime())}] discord.gift/{code} | valid"
+                    )
+                    open("./data/valid.txt", "a").write(f"{code}\n")
+                if req == 404:
+                    print(
+                        f"{Fore.RED}[{strftime('%H:%M', localtime())}] discord.gift/{code} | invalid"
+                    )
+
+                if req == 429:
+                    print(
+                        f"{Fore.YELLOW}[{strftime('%H:%M', localtime())}] discord.gift/{code} | ratelimited"
+                    )
+
+            except Exception as e:
+                print(f"{Fore.RED}[{strftime('%H:%M', localtime())}] {e}")
+
+        print(
+            f"{Fore.LIGHTMAGENTA_EX}[{strftime('%H:%M', localtime())}] Successfully checked {this.codes} codes."
+        )
+        sleep(1.5)
+        os.system("clear")
+
+
+if __name__ == "__main__":
+    while True:
+        code_type = input(
+            f"{Fore.LIGHTMAGENTA_EX}[{strftime('%H:%M', localtime())}] Code Type (boost, classic): "
+        )
+        prox = input(
+            f"{Fore.LIGHTMAGENTA_EX}[{strftime('%H:%M', localtime())}] Proxies (True, False): "
+        )
+        codes = input(
+            f"{Fore.LIGHTMAGENTA_EX}[{strftime('%H:%M', localtime())}] Number of codes: "
+        )
+        SapphireGen(code_type, prox, codes).generate(scrape=False)
```

Comment:
wouldn't it be better if the proxy scrape was part of the input?
like:

```py
scrape_proxy = False

prox = input(
    f"{Fore.LIGHTMAGENTA_EX}[{strftime('%H:%M', localtime())}] Proxies (True, False): "
)
if prox == True:
    scrape_proxy = input(
        f"{Fore.LIGHTMAGENTA_EX}[{strftime('%H:%M', localtime())}] Scrape proxies? (True, False): "
    )


SapphireGen(code_type, prox, codes).generate(scrape=scrape_proxy)
```

Reply:
I did, I must've not have updated the code when I made the pr.